xbout/geometries.py

```diff
@@ -141,8 +144,10 @@ def add_toroidal_geometry_coords(ds, *, coordinates=None, grid=None):
 
     # Add 1D Orthogonal Toroidal coordinates
     ny = ds.dims[coordinates['y']]
-    theta = xr.DataArray(np.linspace(start=0, stop=2 * np.pi, num=ny),
-                         dims=coordinates['y'])
+    # dy should always be constant in x, so it is safe to slice to x=0
```

Comment:
This is for a particular geometry, right? This isn't necessarily true in general

Reply:
My understanding of how we handle coordinate systems is:
* 'x' and 'y' have to be coordinates of the grid, i.e. rows/columns of grid points are along lines of constant y or x. Any more general non-uniformity (i.e. y-direction spacing that depends on x or x-direction spacing that depends on y) has to be handled by the metric.
* therefore `dx` has to be constant in the y-direction, and `dy` has to be constant in the x-direction
    * at least within connected regions - we can have different `dx` in the PFR and the core at the 'same' x-value because they are not connected, but within core, SOL or PFR `dx` has to be constant in the y-direction. There are no branch-cuts in the x-direction, so `dy` has to be constant in the x-direction globally, because along the x-direction all regions are connected.

(@bendudson is the above correct?)

So 'dy should always be constant in x' is true for all topologies currently supported by BOUT++, but we could imagine allowing more general topology at some point... probably a better comment is needed, I'll update.

Reply:
Hi @johnomotani yes I think that is right, or at least I haven't come across a case where it's not true.